pkg/system/phase2_creating.go

```diff
@@ -987,8 +998,13 @@ func (r *Reconciler) UpgradeMigrateDB() error {
 
 // SetDesiredJobUpgradeDB updates the UpgradeJob as desired for reconciling
 func (r *Reconciler) SetDesiredJobUpgradeDB() error {
+	backoffLimit := upgradeJobBackoffLimit
 	r.UpgradeJob.Spec.Template.Spec.Containers[0].Image = r.NooBaa.Status.ActualImage
 	r.UpgradeJob.Spec.Template.Spec.Containers[0].Command = []string{"/noobaa_init_files/noobaa_init.sh", "db_migrate"}
+	// setting the restart policy to never to keep the pods around after failed migrations
+	// also reducing the backoff limit to avoid to many pods staying around in case of an issue
+	r.UpgradeJob.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyNever
+	r.UpgradeJob.Spec.BackoffLimit = &backoffLimit
 	r.setDesiredCoreEnv(&r.UpgradeJob.Spec.Template.Spec.Containers[0])
```

Comment:
move this before the added code (line 1004) - so the two parts will be better divided